gcip/addons/container/jobs/kaniko.py

```diff
@@ -145,24 +124,17 @@ def execute(
         if image_tag and (image_tag == "main" or image_tag == "master"):
             executor_cmd.append(f"--destination {registry}/{image_name}:latest{build_target_postfix}")
 
-    if ecr_login:
-        job.prepend_scripts('mkdir -p /kaniko/.docker && echo "{\\"credsStore\\":\\"ecr-login\\"}" > /kaniko/.docker/config.json')
-
-    if registry_user_env_var and registry_login_env_var:
-        job.prepend_scripts(
-            'mkdir -p /kaniko/.docker && echo "{\\"auths\\":{\\"https://index.docker.io/v1/\\":{\\"username\\":\\"$'
-            + registry_user_env_var
-            + '\\",\\"password\\":\\"$'
-            + registry_login_env_var
-            + '\\"}}}" > /kaniko/.docker/config.json'
-        )
+    job = Job(
+        script=" ".join(executor_cmd),
+        namespace="build_container_image",
```

Comment:
What about `namespace=build` and `name=kaniko`?
We are doing the same with all python jobs:
https://github.com/dbsystel/gitlab-ci-python-library/blob/main/gcip/addons/python/jobs.py

I know that we force the job to run at max in parallel with all other jobs in the `build` jobs. The current implementation would force to run the job at max in the unique stage `build_container_image`.

The best solution would be to create jobs without any name and namespace set up. However this requires a fundamental change in the gcip. So currently I would advice to set namespace and name to a similar scheme than in other jobs (e.g. python).

Does that sound reasonable?

Reply:
Yes, but think about the gcip build process. 
If namespace for the build pipeline of the gcip python package and the container build are the same. It could lead to a race condition. Means the container build wants to load the python package which is build inside the same stage. 
If the container built is faster than the python package build (which is reasonable), the container build will fail because the python package is not ready. 

I understand an appreciate your suggestion. Maybe my concerns are not true, please let me know!

P.s: Do you have an idea, how to implement it in a better way inside the core of the gcip?

Reply:
> Maybe my concerns are not true, please let me know!

The user could decide in which "final" stage the job will run

```
pipeline.add_children(bdist_whee_job, "python") # would run in 'build_python
pipeline.add_children(kaniko_job, "docker")     # would run in 'build_docker
```

See https://dbsystel.github.io/gitlab-ci-python-library/user/index.html#_reuse_jobs

----

However it would be best, if the user could fully decide about the name and namespace of predefined jobs. That is currently not possible and for sure out of scope of this MR

-->

> P.s: Do you have an idea, how to implement it in a better way inside the core of the gcip?

Yes, exactly as I stated in my comment before:

> The best solution would be to create jobs without any name and namespace set up. However this requires a fundamental change in the gcip.

Reply:
Ok, i did change the namspace of the kaniko job to `build`.